sklearn/preprocessing/label.py

```diff
@@ -795,7 +797,14 @@ def _transform(self, y, class_mapping):
         indices = array.array('i')
         indptr = array.array('i', [0])
         for labels in y:
-            indices.extend(set(class_mapping[label] for label in labels))
+            index = set()
+            for label in labels:
+                try:
+                    index.add(class_mapping[label])
+                except KeyError:
+                    warnings.warn("Unkown class {0!r} will be ignored"
```

Comment:
Sorry tofuss over this again, but perhaps it's better still to accumulate all the unknown labels and warn at the end

Reply:
Why? I disagree, there is no internal purpose for the accumulation of these unknown classes. The warning is sufficient for the user to go and double check his/her data.